### ISSUE TITLE
Empty lines around examples, hooks, example groups

### ISSUE THREAD

**Issue by backus** [rubocop-hq/rubocop-rspec#272]
I'm often surprised when I see specs written like so

```ruby
describe 'foo' do
  subject { thing }
  let(:foo) { bar }
  let(:bar) { baz }
  before { some_setup }
  after { some_cleanup }
  it 'does x' do
    blah
  end
  it 'does y' do
    blah
  end
end

```

I think it is reasonable to say that this should **at least** be written like so:

```ruby
describe 'foo' do
  subject { thing }

  let(:foo) { bar }
  let(:bar) { baz }

  before { some_setup }
  after { some_cleanup }

  it 'does x' do
    blah
  end

  it 'does y' do
    blah
  end
end
```

This isn't without precedent too. If I wrote code like this:

```ruby
module Foo
  def bar
  end
  def baz
  end
  private
  def qux
  end
end
```

Rubocop would flag it

```
ex.rb:4:3: C: Style/EmptyLineBetweenDefs: Use empty lines between method definitions.
  def baz
  ^^^
ex.rb:6:3: C: Style/EmptyLinesAroundAccessModifier: Keep a blank line before and after private.
  private
  ^^^^^^^
```

**Comment by backus** [comment]
@andyw8 @bquorning what do you think?

**Comment by andyw8** [comment]
Should RuboCop's `EmptyLinesAroundBlockBody` already handle this? (can't check at the moment)

**Comment by backus** [comment]
@andyw8 I don't think so because the code base that I most recently saw this in was in rubocop itself

**Comment by andyw8** [comment]
Hmm, looks like the default is `no_empty_lines`?

https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L438

**Comment by backus** [comment]
@andyw8 interesting. I set that config and ran it on my example and it produced this

```ruby
describe 'foo' do

  subject { thing }
  let(:foo) { bar }
  let(:bar) { baz }
  before { some_setup }
  after { some_cleanup }
  it 'does x' do

    blah

  end
  it 'does y' do

    blah

  end

end
```

heh

**Comment by andyw8** [comment]
Ah, looking at [the examples](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/EmptyLinesAroundBlockBody) I misunderstood this.

I actually think a general "whitespace around multiline blocks" cop for RuboCop would be useful, but it might be easier to implement an RSpec-only cop here to begin with.

**Comment by backus** [comment]
Here is a WIP if you are interested https://github.com/backus/rubocop-rspec/pull/273

**Comment by Darhazer** [comment]
I would love if there is an option not to require blank lines between single-line `it` examples. Here is how we test validations:

```
describe 'validations' do
    subject { build_stubbed :user }

    it { is_expected.to validate_presence_of(:first_name) }
    it { is_expected.to validate_presence_of(:last_name) }
    it { is_expected.to validate_presence_of(:phone_number) }
    ...
end
```

**Comment by Darhazer** [comment]
We already have a requirement for empty line after the final let; after the subject; after a hook and after an example group. This covers most of this. The only thing left that I can see is empty line after example, which I would like to see configurable regarding the single-line examples.

**Comment by Darhazer** [comment]
The last bit was fixed in #823
